prody/dynamics/analysis.py

```diff
@@ -748,36 +698,121 @@ def detectHinges(v, threshold):
 
         ### Select hinges from regions ###
         fil=[]
-        n, _ = vecs.shape
-        
         for reg in merged:
-            s0, s1 = reg[0], reg[-1]
-
-            if len(reg) < n // 10 and v[s0] * v[s1] > 0:
-                continue
-            
             av = np.abs(v[reg])
             
-            if len(reg) >= s + 4:
-                fil.extend([reg[i] for i in np.argsort(av)[:2]])
+            if len(reg) >= s + 4: # region with length of 2+ discrete regions
+                fil.extend([reg[i] for i in np.argsort(av)[:2] if av < band]) # end hinges within band
             else:
-                fil.append(reg[np.argmin(av)])
+                if v[reg[0]] * v[reg[-1]] > 0 : # skip transient region
+                    continue
+                fil.append(reg[np.argmin(av)]) # minimum point as hinge
 
         return [int(x) for x in fil]
+    
+
+def _getModeVecs(gnm, n_modes=None, min_variance=0.33, atoms=None):
+    """
+    Return eigenvector array for hinge detection.
+
+    Handles GNM and ModeSet inputs, including reuse of precomputed modes.
+    """
+    if isinstance(gnm, ModeSet):
+        return gnm.getArray()
+
+    if not isinstance(gnm, GNMBase):
+        raise TypeError("Input must be a GNM model or ModeSet instance.")
```

Comment:
Maybe allow gnm=None if atoms is not None as a new gnm can be instantiated? People are unlikely to input a newly instantiated gnm object themselves